src/libs/actions/Report.js

```diff
@@ -432,6 +432,67 @@ function unsubscribeFromReportChannel(reportID) {
     Pusher.unsubscribe(pusherChannelName);
 }
 
+/**
+ * Get the report ID, and then the actions, for a chat report for a specific
+ * set of participants
+ *
+ * @param {String[]} participants
+ */
+function fetchOrCreateChatReport(participants) {
+    let reportID;
+
+    if (participants.length < 2) {
+        throw new Error('fetchOrCreateChatReport() must have at least two participants');
+    }
+
+    API.CreateChatReport({
+        emailList: participants.join(','),
+    })
+
+        .then((data) => {
+            if (data.jsonCode !== 200) {
+                alert(data.message);
+                return;
```

Comment:
Promise chains are tricky, and so I don't think this `return` is doing the right thing. All it's doing is returning `null` from this part of the promise chain which gets passed to the next `.then()` in the chain. This means the next `.then()` is still going to run, it's just that `data` is going to be `null` which will cause some errors in the code (I'll point out below).

I think you want to cancel the entire chain in this `if` statement by throwing an error: `throw new Error(data.message);`. This will stop all other `.then()` from being ran, and it will trigger the next `catch()` in the promise chain if there is one. Since there is no `catch()` in this chain, nothing would happen, which is fine.